Catappult/appcoins-wallet-android
# [WAL-110] Lock promotions for users who haven't made a purchase

**What does this PR do?**

Adds the layout and the logic to not show promotions to users until they make their first purchase.
Changed the logic of when to display the promotions badge
Added logic to retrievePromotions onResume so that if the user comes from the game directly into the promotions screen he is able to see the promotions.

**Database changed?**

   Yes, added new table for the walletOrigin

**Where should the reviewer start?**

PromotionsPresenter.kt

**How should this be manually tested?**

Check if before doing a purchase the user doesn't see any promotions and after doing one the promotions show up.
Also, check if no promotion is shown (this is more of an API test, so if you don't have one, no need to do it) for some cases as specified in the ticket.

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/WAL-110

**Code Review Checklist**
